src/Development/IDE/GHC/Util.hs

```diff
@@ -166,16 +167,20 @@ moduleImportPath (takeDirectory . fromNormalizedFilePath -> pathDir) mn
 --   if they are created with the same call to 'newHscEnvEq'.
 data HscEnvEq
     = HscEnvEq !Unique !HscEnv
+               [(InstalledUnitId, DynFlags)] -- In memory components for this HscEnv
+               -- This is only used at the moment for the import dirs in
+               -- the DynFlags
     | GhcVersionMismatch { compileTime :: !Version
```

Comment:
We can make this GhcVersionMismatch into a proper diagnostic now I think rather than this special constructor on `HscEnvEq`. Not sure whether to do that in this patch.

Reply:
I have another PR making changes to `HscEnvEq` which will need to be reviewed once this is merged, and I am happy to convert `GhcVersionMismatch` to a proper diagnostic if that is possible at all

Reply:
Would you prefer me to remove it in this patch? Or leave it until later?